application/event_processor/dos.py

```diff
@@ -39,86 +43,127 @@ class DoSService:
-        return self.odscode[0:5]
-
-    def get_changes(self, nhs_entity) -> dict:
+    def get_changes(self, nhs_entity: NHSEntity) -> dict:
         """Returns a dict of the changes that are required to get
         the service inline with the given nhs_entity
         """
-
         changes = {}
+        add_field_to_change_request_if_not_equal(changes, "website", self.web, nhs_entity.Website)
+        add_field_to_change_request_if_not_equal(changes, "postcode", self.postcode, nhs_entity.Postcode)
+        add_field_to_change_request_if_not_equal(changes, "phone", self.publicphone, nhs_entity.Phone)
+        add_field_to_change_request_if_not_equal(changes, "publicname", self.publicname, nhs_entity.OrganisationName)
+        add_address_to_change_request_if_not_equal(changes, "address", self.address, nhs_entity)
+        return changes
 
-        # WEBSITE
-        if self.web != nhs_entity.Website:
-            changes["Website"] = nhs_entity.Website
 
-        # TODO in future tickets: Add in checks for the rest of the
-        # possible changes
+def add_field_to_change_request_if_not_equal(changes: dict, change_key: str, dos_value: str, nhs_uk_value: str) -> dict:
+    """Adds field to the change request if the field is not equal
+    Args:
+        changes (dict): Change Request changes
+        change_key (str): Key to add to the change request
+        dos_value (str): Field from the DoS database for comparision
+        nhs_uk_value (str): NHS UK Entity value for comparision
 
-        return changes
+    Returns:
+        dict: Change Request changes
+    """
+    logger.debug(f"Comparing {change_key} values")
+    if compare_values(str(dos_value), str(nhs_uk_value)):
+        changes[change_key] = nhs_uk_value
+    return changes
 
 
-def get_matching_dos_services(odscode):
-    """Retrieves DoS Services from DoS database
+def add_address_to_change_request_if_not_equal(
+    changes: dict, change_key: str, dos_address: str, nhs_uk_entity: NHSEntity
+) -> dict:
+    """Adds the address to the change request if the address is not equal
+
+    Args:
+        changes (dict): Change Request changes
+        change_key (str): Key to add to the change request
+        dos_address (str): Address from the DoS database for comparision
+        nhs_uk_entity (NHSEntity): NHS UK Entity for comparision
+
+    Returns:
+        dict: Change Request changes
+    """
+    nhs_uk_address_lines = [
+        nhs_uk_entity.Address1,
+        nhs_uk_entity.Address2,
+        nhs_uk_entity.Address3,
+        nhs_uk_entity.City,
+        nhs_uk_entity.County,
+    ]
+    nhs_uk_address = [address for address in nhs_uk_address_lines if address is not None and address.strip() != ""]
+    nhs_uk_address_string = "$".join(nhs_uk_address)
+    if compare_values(dos_address, nhs_uk_address_string):
+        changes[change_key] = nhs_uk_address
+    return changes
+
 
-    input:  odscode
+def compare_values(dos_value: str, nhs_uk_value: str) -> bool:
```

Comment:
Whats the reason for this function to exist?

Reply:
It reduces code duplication and adds logging

Reply:
i would be inclined to remove it and just to x != y in line, the only bit of duplication would be the logging but if you do that further up the stack you can add more content, e.g. f`{WEBSITE CHANGE KEY} are not equal, dos value ${dos.website} (dos value)  != ${nhsuk.website} (nhs value}` for example, the problem with the current name of this function is you dont know form the name how its going to compare, if you read x != y the logical operation is plain to see if you just see compare_values you dont know and probably wouldnt expect it to return true for a negative comparison